src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,52 @@
+package duke.command;
+
+import duke.data.TaskList;
+import duke.data.task.Task;
+import duke.exception.DeadlineMissingTagException;
+import duke.exception.DeadlineMissingDescriptionException;
+import duke.exception.DeadlineMissingTimeException;
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+public class DeadlineCommand extends AddCommand {
+    public static final String COMMAND_WORD = "DEADLINE";
+    private String deadlineTime;
+
+    public DeadlineCommand() {
+        super();
+    }
+
+    @Override
+    public void checkAndSetParameters(String parameterInput) throws DukeException {
+        checkParameters(parameterInput);
+        setParameters(parameterInput);
+    }
+
+    @Override
+    protected void checkParameters(String parameterInput) throws DukeException {
+        if (!containsTimeSeparator(parameterInput, " /by ")) {
```

Comment:
Consider making `" /by "` into a constant, especially since it includes the whitespace, which can be easily mistakenly written.

Reply:
Now added as a constant of the class:
`    public static final String TIME_SEPARATOR = " /by ";`